hugegraph-core/src/main/java/com/baidu/hugegraph/backend/store/BackendMutation.java

```diff
@@ -111,6 +111,19 @@ private void optimizeUpdates(BackendEntry entry, Action action) {
                                   " transaction between %s and %s",
                                   entry, originItem.entry()));
                     }
+
+                    if (originAction == Action.INSERT ||
+                        originAction == Action.DELETE) {
+                        throw incompatibleActionException(action, originAction);
+                    } else {
+                        Id subId = entry.subId();
+                        Id originSubId = originItem.entry().subId();
+                        assert subId != null;
+                        if (subId == originSubId || subId.equals(originSubId)) {
+                            iter.remove();
+                        }
```

Comment:
just keep lines 119-124

Reply:
if only keep lines 119-124
`@Test
    public void testDeleteAndAppendEntry() {
        BackendMutation mutation = new BackendMutation();
        BackendEntry entry1 = constructBackendEntry("1");
        BackendEntry entry2 = constructBackendEntry("1", "name", "marko");

        mutation.add(entry1, Action.DELETE);
        Assert.assertThrows(HugeException.class, () -> {
            mutation.add(entry2, Action.APPEND);
        });
    }`

these scripts are probably not able to pass

Reply:
OK